packages/did-resolver/index.ts

```diff
@@ -103,18 +109,30 @@ function isFlickrBase58(candidate: string) {
   return candidate.length === translator.maxLength && candidate.match(new RegExp(`^[${translator.alphabet}]+$`));
 }
 
+function destructure(data: string) {
+  let regex = new RegExp(`^([a-zA-Z0-9])(${Object.values(DID_TYPE_TO_SHORT_TYPE).join('|')})(.*)`);
+  let match = data.match(regex);
+  if (match === null) {
+    throw new Error(`Cannot destrucure data "${data}""`);
+  } else {
+    let versionChar = match[1];
+    let shortType = match[2];
+    let uniqueId = match[3];
+    let type = SHORT_TYPE_TO_DID_TYPE[shortType];
+    if (!type) {
+      throw new Error(`Invalid DID identifier: unknown type "${data[1]}"`);
+    }
+    return { version: versionFromChar(versionChar), type, uniqueId };
+  }
+}
+
 export function parseIdentifier(identifier: string): CardstackIdentifier {
   let data = identifier.substring(0, identifier.length - HASH_LENGTH);
   let checksum = identifier.substring(identifier.length - HASH_LENGTH, identifier.length);
   if (hashFunc(data) !== checksum) {
     throw new Error('Invalid DID identifier: checksum failed');
   }
-  let version = versionFromChar(data[0]);
-  let type = SHORT_TYPE_TO_DID_TYPE[data[1]];
```

Comment:
@lukemelia is this a strict restriction on the cardstack method identifier. My property test failed because there was ["cs" short type](https://github.com/cardstack/cardstack/blob/3590a2666d4c7f3f3663cf36fdb81392cb050b1a/packages/did-resolver/index.ts#L17), which is two characters long. 

Initially, I thought this was a mistake but then I realise you specify within the README that this short-type is strictly one character

Reply:
cc @jurgenwerk

Reply:
I am going with the direction of using single character to represent our did type (That means migrating  all s3 docs which have did wrong)

For future reference. Test on python fail

```
Error: Invalid DID identifier: unknown type "c"
```